desmume/src/frontend/posix/gtk/main.cpp

```diff
@@ -1779,6 +1795,131 @@ static void PlayMovieDialog(GSimpleAction *action, GVariant *parameter, gpointer
     g_object_unref(pFileSelection);
 }
 
+static void ImportBackupMemoryDialog(GSimpleAction *action, GVariant *parameter, gpointer user_data)
+{
+    GtkFileFilter *pFilter_raw, *pFilter_ar, *pFilter_any;
+    GtkWidget *pFileSelection;
+    GtkWidget *pParent;
+    gchar *sPath;
+
+    if (desmume_running())
+        Pause(NULL, NULL, NULL);
+
+    pParent = GTK_WIDGET(pWindow);
+
+    pFilter_raw = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_raw, "*.sav");
+    gtk_file_filter_set_name(pFilter_raw, "Raw/No$GBA Save format (*.sav)");
+
+    pFilter_ar = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_ar, "*.duc");
+    gtk_file_filter_add_pattern(pFilter_ar, "*.dss");
+    gtk_file_filter_set_name(pFilter_ar, "Action Replay DS Save (*.duc,*.dss)");
+
+    pFilter_any = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_any, "*.sav");
+    gtk_file_filter_add_pattern(pFilter_any, "*.duc");
+    gtk_file_filter_add_pattern(pFilter_any, "*.dss");
+    gtk_file_filter_set_name(pFilter_any, "All supported types");
+
+    /* Creating the selection window */
+    pFileSelection = gtk_file_chooser_dialog_new("Import Backup Memory From ...",
+            GTK_WINDOW(pParent),
+            GTK_FILE_CHOOSER_ACTION_OPEN,
+            GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
```

Comment:
Please also avoid using `GtkStock` symbols, as those are deprecated.

I tried to avoid having any deprecated symbol to ease GTK 4 porting.  I already have a [working branch](https://github.com/linkmauve/desmume/tree/gtk4) for it, I’m just waiting for it to get a proper release before switching.

Maybe it’d be better to `#define GTK_DISABLE_DEPRECATED` somewhere, to avoid people adding new deprecated symbols.

Reply:
There, as of 64da66f37b245dccac694096b47648af16dd2339 using deprecated symbols will be an error instead of a warning.